fix: fix linting errors in the samples

DominicKramer opened the pull request:
`eslint` was reporting the following error:
```
error  Async functions are not supported until Node.js 7.6.0. The configured version range is '>=6.0.0'  node/no-unsupported-features/es-syntax
```
This happened because `eslint` has a new `node/no-unsupported-features/es-syntax` that by default reports an error.

The error occurs because the samples use async/await, async/await is not available until Node >= 7.6.0, and the samples `package.json` specifies only Node >= 6.0 is required.

To fix the linting errors, the `node/no-unsupported-features/es-syntax` was configured to only issue a warning.

This was done, instead of updating the `package.json` to specify that Node >= 7.6.0 is required,   since the library `@google-cloud/logging-bunyan` library only requires Node >= 6.0.0.  It is only if you want to use async/await that Node >= 7.6.0 is required.

ofrobots commented:
Also, perhaps another way to deal with this is adjusted eslint config that makes the offending errors into warnings?
/cc @JustinBeckwith

DominicKramer commented:
@kjin The samples `package.json` specified `node@>=6.0.0`.  However, the samples use async/await.  As a result, eslint reported an error since async/await is only available in `node@>= 7.6.0`.  Specifying the samples are for `node@>=7.6.0` resolves the issue.

DominicKramer commented:
I've updated the PR to have the `node/no-unsupported-features/es-syntax` rule issue a warning instead of an error instead of updating the required Node.js version in `package.json`, and I updated the PR abstract with a description of why that choice was made.
